dask/dataframe/shuffle.py

```diff
@@ -497,14 +497,15 @@ def set_index(
             sizes = []
 
         divisions = index2._repartition_quantiles(npartitions, upsample=upsample)
-        iparts = index2.to_delayed(optimize_graph=False)
-        mins = [ipart.min() for ipart in iparts]
-        maxes = [ipart.max() for ipart in iparts]
+        mins = index2.map_partitions(operator.methodcaller("min"))
+        maxes = index2.map_partitions(operator.methodcaller("max"))
```

Comment:
```suggestion
        mins = index2.map_partitions(M.min)
        maxes = index2.map_partitions(M.max)
```

The `M` object is slightly easier to serialize